pilot/pkg/proxy/envoy/v2/util.go

```diff
@@ -115,3 +113,21 @@ func buildProtoStruct(name, value string) *google_protobuf.Struct {
 		},
 	}
 }
+
+// messageToStruct is the most inefficient way to pass a struct, but will do for first
+// iteration.
+func messageToStruct(msg proto.Message) *types.Struct {
+	buf := &bytes.Buffer{}
+	if err := (&jsonpb.Marshaler{OrigName: true}).Marshal(buf, msg); err != nil {
+		log.Error(err.Error())
```

Comment:
This is a `warn` at best, but probably just `debug`: we should be able to guarantee this never actually happens at runtime. Also, here's a more compact way to implement this:

```go
func MessageToStruct(p proto.Message) (s *types.Struct, err error) {
	bytes, err := proto.Marshal(p)
	if err != nil {
		return nil, err
	}
	err = s.Unmarshal(bytes)
	return s, err
}
```